File: problem/lcm/6/main.py

# main.py: main file for an "lcm" solution
#
# This is appended to the submitted Python program and tests the lcm function.
#
# The submitted file processes the standard input and this processes the
# "in.txt" file.
#
#
# Copyright (C) 2020-2021  Rudy Matela

# standard input is already processed
# as the submitted programmed is run when imported
with open("in.txt") as filein:
  for line in filein:
    x,y = [int(x) for x in line.split()]
    print(lcm(x,y))

File: problem/lcm/6/main.cs
// main.cs: main file for solutions to the lcm exercise
//
// This is to be linked to the submitted file.
// It processes values from standard input then from the "in.txt" file.
//
//
// Copyright (C) 2020-2021  Rudy Matela
// All rights reserved

using System;

class TheNewMain
{
	static void Main()
	{
		string line;

		// processes standard input
		while ((line = Console.ReadLine()) != null) {
			Solve1(line);
		}

		// processes additional inputs from in.txt
		var file = new System.IO.StreamReader("in.txt");
		while ((line = file.ReadLine()) != null) {
			Solve1(line);
		}
		file.Close();
	}

	static void Solve1(string line) {
		string[] inputs = line.Split();
		int a = Convert.ToInt32(inputs[0]);
		int b = Convert.ToInt32(inputs[1]);
		Console.WriteLine(Program.LCM(a,b));
	}
}
